src/runtime/hexagon_remote/elf.h

```diff
@@ -143,7 +159,8 @@ struct elf_t {
         // stuff, the same size again to make a writeable copy, and
         // then an extra page for the global offset table for PIC
         // code.
-        buf = (char *)mmap(NULL, size * 2 + global_offset_table_size,
+        mmap_ptr = (mmap_ptr_fn) halide_get_symbol("mmap");
```

Comment:
This complexity is unnecessary, if the symbol is weak, I believe you can simply do:

if (mmap) {
    buf = (char *)mmap(...);
} else {
   // symbol isn't there, error
}